sound/soc/intel/boards/sof_sdw.c

```diff
@@ -1303,25 +1303,24 @@ static int get_slave_info(const struct snd_soc_acpi_link_adr *adr_link,
 	}
 
 	/* gather other link ID of slaves in the same group */
-	for (adr_next = adr_link + 1; adr_next && adr_next->num_adr;
-		adr_next++) {
-		const struct snd_soc_acpi_endpoint *endpoint;
-
-		endpoint = adr_next->adr_d->endpoints;
-		if (!endpoint->aggregated ||
-		    endpoint->group_id != *group_id)
-			continue;
-
-		if (index >= SDW_MAX_CPU_DAIS) {
-			dev_err(dev, " cpu_dai_id array overflows");
-			return -EINVAL;
-		}
+	for (adr_next = adr_link + 1; adr_next && adr_next->num_adr; adr_next++) {
+		unsigned int link_codecs = 0;
```

Comment:
Humm, the commit message is confusing.

Aggregation means we need to use multiple SoundWire links. the typical example is 2 amplifiers that constitute a logical group even if they are on two links. But when we have 2 amplifiers on the link we don't treat them as a logical group, because there's no need to do anything.

In addition, we could have 2 links with one amp and one mic each, with 2 groups spanning two links each.

So we should describe the cases we want to handle more clearly in the commit message. Which of the 3 is this trying to address?

Reply:
I will attempt to reword the commit message a little, the primary bit I am driving at here is basically this snippet here:

```c
endpoint = adr_next->adr_d->endpoints;
if (!endpoint->aggregated || endpoint->group_id != *group_id)
	continue;
```

That looks at the first endpoint on the first device on the link and skips the whole link if it doesn't match the group we are looking for. However, there may be other devices on that link that do have endpoints in the group we are looking for. So for example say I have a system like:

SDW0 -> Amp 1
SDW1 -> Codec, Amp 2

Where the two amps are aggregated, the current code would see the non-aggregate Cohen at the start of SDW1 and not process the amp we want aggregated.

I assume we don't have any systems like this already, but fixing it will both support systems like that in the future and facilitate making the code simpler now.